api/envoy/service/discovery/v3/discovery.proto

```diff
@@ -107,6 +128,22 @@ message DiscoveryResponse {
 
   // The control plane instance that sent the response.
   config.core.v3.ControlPlane control_plane = 6;
+
+  // [#not-implemented-hide:]
+  // An optional per-resource status response that can be used when no matching resource(s)
+  // are sent to the client, and will describe the reason. This will be a map
+  // from a resource name to its status. A resource that doesn't appear in the map
+  // implies the resource response is valid. No status map implies that all the
+  // resources in the response are valid.
+  map<string, DiscoveryResponseStatus> resources_status = 7;
+
+  // [#not-implemented-hide:]
+  // An optional API patch version of the requested resource that is supported by
+  // the client. This is useful for use-cases where the client and server support new
+  // API features, before the next API minor version is rolled out.
+  // If not included, the patch version defaults to 0, stating that the client
+  // supports API features up to the client's minor version.
+  uint32 patch_version = 8;
```

Comment:
Why is this in response? Shouldn't it only be request?

Reply:
Moved to `DiscoveryResponse`.